### DISCUSSION TITLE
Use case for creating a stream with start & end time in the past

### DISCUSSION THREAD

**Discussion by partylikeits1983** [sablier-labs/evm-monorepo#1111] [Q&A] [answered]
I was wondering why does Sablier allow for token streams to be created with a `startTime` that is in the past relative to the current `block.timestamp`? Are there really use cases to create a stream that has a start and end time that are in the past relative to the current timestamp? 

[This function](https://github.com/sablier-labs/v2-core/blob/43cf7c9d968e61a5a03e9237a71a27165b125414/src/SablierV2LockupLinear.sol#L163) doesn't check if startTime is less than the current timestamp. This doesn't have adverse side effects, just a curious why isn't there a check for: 
```solidity
require(startTime >= block.timestamp)
```

What use cases are there for creating a stream in the past? 

Source: https://github.com/sablier-labs/v2-core/blob/43cf7c9d968e61a5a03e9237a71a27165b125414/src/SablierV2LockupLinear.sol#L163

**Comment by smol-ninja** [comment, marked as answer]
Hi @partylikeits1983 thanks for asking the question.

1. Allowing `startTime` to be in the past: so that you can create backdated vesting streams. There could be a use case such as having prior commitment to an investor but decides to create a stream in the future. This would require start time to be in the past.
2. Allowing `endTime` to be in the past (this will be enabled in the future release): you can read about it [here](https://github.com/sablier-labs/v2-core/discussions/997).

Let me know if this answers your questions. Feel free to ask me if you have more.

**Reply by PaulRBerg** [reply]
> having prior commitment to an investor but decides to create a stream in the futur

Basically: retroactive vesting.
